integration-test/src/test/java/org/apache/gravitino/integration/test/authorization/AccessControlIT.java

```diff
@@ -0,0 +1,260 @@
+    Assertions.assertEquals(group.name(), groupName);
+    Assertions.assertTrue(group.roles().isEmpty());
+
+    // Get a not-existed group
+    Assertions.assertThrows(NoSuchGroupException.class, () -> metalake.getGroup("not-existed"));
+
+    Assertions.assertTrue(metalake.removeGroup(groupName));
+    Assertions.assertFalse(metalake.removeGroup(groupName));
+  }
+
+  @Test
+  void testManageRoles() {
+    String roleName = "role#123";
+    Map<String, String> properties = Maps.newHashMap();
+    properties.put("k1", "v1");
+    SecurableObject metalakeObject =
+        SecurableObjects.ofMetalake(
+            metalakeName, Lists.newArrayList(Privileges.CreateCatalog.allow()));
+    Role role = metalake.createRole(roleName, properties, Lists.newArrayList(metalakeObject));
+
+    Assertions.assertEquals(roleName, role.name());
+    Assertions.assertEquals(properties, role.properties());
+
+    // Verify the object
+    Assertions.assertEquals(1, role.securableObjects().size());
+    SecurableObject createdObject = role.securableObjects().get(0);
+    Assertions.assertEquals(metalakeObject.name(), createdObject.name());
+    Assertions.assertEquals(metalakeObject.type(), createdObject.type());
+
+    // Verify the privilege
+    Assertions.assertEquals(1, createdObject.privileges().size());
+    Privilege createdPrivilege = createdObject.privileges().get(0);
+    Assertions.assertEquals(createdPrivilege.name(), Privilege.Name.CREATE_CATALOG);
+    Assertions.assertEquals(createdPrivilege.condition(), Privilege.Condition.ALLOW);
+
+    // Get a role
+    role = metalake.getRole(roleName);
+
+    Assertions.assertEquals(roleName, role.name());
+    Assertions.assertEquals(properties, role.properties());
+
+    // Verify the object
+    Assertions.assertEquals(1, role.securableObjects().size());
+    createdObject = role.securableObjects().get(0);
+    Assertions.assertEquals(metalakeObject.name(), createdObject.name());
+    Assertions.assertEquals(metalakeObject.type(), createdObject.type());
+
+    // Verify the privilege
+    Assertions.assertEquals(1, createdObject.privileges().size());
+    createdPrivilege = createdObject.privileges().get(0);
+    Assertions.assertEquals(createdPrivilege.name(), Privilege.Name.CREATE_CATALOG);
+    Assertions.assertEquals(createdPrivilege.condition(), Privilege.Condition.ALLOW);
+
+    // Delete a role
+    Assertions.assertTrue(metalake.deleteRole(roleName));
+    Assertions.assertFalse(metalake.deleteRole(roleName));
+  }
+
+  @Test
+  void testManageUserPermissions() {
+    String username = "user1#123";
+    User user = metalake.addUser(username);
+    Assertions.assertTrue(user.roles().isEmpty());
+
+    String roleName = "role#123";
+    Map<String, String> properties = Maps.newHashMap();
+    properties.put("k1", "v1");
+    SecurableObject metalakeObject =
+        SecurableObjects.ofMetalake(
+            metalakeName, Lists.newArrayList(Privileges.CreateCatalog.allow()));
```

Comment:
Will you check if the metadata object exist or not? If so, please add the test to cover this.

Reply:
Added the test case.